samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java

```diff
@@ -222,8 +222,7 @@ private     void                notifyInstructor                    (String site
             Site            site                    = siteService.getSite(siteID);
             Set<String>     siteUsersHasRole;
             
-            if (replacementValues.get("releaseToGroups") != null){
-            	siteUsersHasRole = extractInstructorsFromGroups(site,replacementValues.get("releaseToGroups") );
+            if ((replacementValues.get("releaseToGroups") != null) && !((siteUsersHasRole=extractInstructorsFromGroups(site,replacementValues.get("releaseToGroups") )).isEmpty())){
```

Comment:
you do two separate if statements to make this more readable

Reply:
Sorry, I don't get it how can I avoid the code duplication with two if statements. The duplicated code is for the two else blocks, but the if blocks are nested. Am I missing something?

Reply:
it something like this: 
if (A) then
{
	if (B) then doX
	else doY
}
else doY

I don't get how can I avoid duplicating doY.

Reply:
https://github.com/sakaiproject/sakai/compare/master...ottenhoff:SAM-3271